src/App.tsx

```diff
@@ -1,96 +1,54 @@
 import React from 'react';
 import './App.css';
-import { getNumbers } from './utils';
+import { useSearchParams } from 'react-router-dom';
+import { getExtremes, getNumbers, getVisibleItems } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemList } from './components/ItemList';
+import { PerPageSelector } from './components/PerPageSelector';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const page = +(searchParams.get('page') || 1);
+  const perPage = +(searchParams.get('perPage') || 5);
```

Comment:
in the real project, we don't write numbers like this
we create constants for it
```
const CURRENT_PAGE_DEFAULT_VALUE = 1;
const PER_PAGE_DEFAULT_VALUE = 5;
```
and we store these variables in a separate file - constants.ts